lib/ssl/src/ssl_connection.erl

```diff
@@ -1179,6 +1179,34 @@ certify(internal, #client_key_exchange{exchange_keys = Keys},
 	#alert{} = Alert ->
 	    handle_own_alert(Alert, Version, ?FUNCTION_NAME, State)
     end;
+%% The response will be handled only when a certificate status request has
+%% been sent by the client and confirmed by the server.
+certify(internal, #certificate_status{response = OcspRespDer},
+        #state{handshake_env = #handshake_env{
+                   ocsp_stapling_state = #{
+                       is_client_hello_requested := true,
+                       is_server_hello_requested := true,
+                       ocsp_responder_certs := ResponderCerts,
+                       ocsp_nonce := OcspNonce} = OcspState} = HsEnv,
+               connection_env = #connection_env{
+                   negotiated_version = Version}} = State,
+        Connection) ->
+    Result = public_key:ocsp_status(
+        OcspRespDer, ResponderCerts, OcspNonce),
+    NewOcspState = OcspState#{
+        ocsp_stapling_result => Result},
+    NewState = State#state{handshake_env =
+        HsEnv#handshake_env{ocsp_stapling_state = NewOcspState}},
```

Comment:
We can not have this check below here!  We need to just save the result in the state here, and all checks should be done during the certificate path_validation in the verify fun. In this first version, there will only ever be one  response,
as you have not implemented multiple responses.

Reply:
I agree with @IngelaAndin and I'd like to add here that I think we can have multiple reponses here even according to rfc6960. Current implementation won't handle responses with multiple SingleReponse elements.
```asn.1
ResponseData ::= SEQUENCE {
      version              [0] EXPLICIT Version DEFAULT v1,
      responderID              ResponderID,
      producedAt               GeneralizedTime,
      responses                SEQUENCE OF SingleResponse,
      responseExtensions   [1] EXPLICIT Extensions OPTIONAL }
```

Reply:
> I agree with @IngelaAndin and I'd like to add here that I think we can have multiple reponses here even according to rfc6960. Current implementation won't handle responses with multiple SingleReponse elements.
> 
> ```asn
> ResponseData ::= SEQUENCE {
>       version              [0] EXPLICIT Version DEFAULT v1,
>       responderID              ResponderID,
>       producedAt               GeneralizedTime,
>       responses                SEQUENCE OF SingleResponse,
>       responseExtensions   [1] EXPLICIT Extensions OPTIONAL }
> ```

I have noticed this, but it is hard to find the response related to the server certificate from a list of responses as the order of the list is related to the order of request which is send from the server and there is no field as I have checked can help to find the right cert.

Reply:
Additional comment about the elements of the response. A server can be configured with multiple certificates and the BasicOCSPReponse will return a SingleReponse for each server cert and we should choose the right one based on which one was chosen for the handshake (server's Certificate message).

Reply:
So we have a new understanding of this. And we should actual  postpone the path_validation to we have this info in the stapling case. And let save this until we prototped the new path_validation handling.

Reply:
> So we have a new understanding of this. And we should actual postpone the path_validation to we have this info in the stapling case. And let save this until we prototped the new path_validation handling.

pending